src/typing/constraint_js.ml

```diff
@@ -529,21 +529,34 @@ let copy_node node = match node with
     Root { rank; constraints = Unresolved (copy_bounds bounds) }
   | _ -> node
 
+type scope_kind =
```

Comment:
Idea: If we called this 'binding_type' instead of 'scope_kind' we could probably get away with creating something like a `ConstantBinding` (and even a `TypeBinding`) -- which would let us get rid of the `for_type` and `constant` flags in all of the utils functions + `scope_entry` record.

This seems like more of a nice-to-have, though, so unless you're just feeling eager it's probably fine to do such a consolidation sometime after this goes in.